route/src/route_generator_worker.cpp

```diff
@@ -438,5 +438,51 @@ namespace route {
         return true;
     }
 
+    bool RouteGeneratorWorker::crosstrack_error_check(const geometry_msgs::PoseStampedConstPtr& msg, lanelet::ConstLanelet current)
+    {
+  
+       lanelet::BasicPoint2d position;
+
+        position.x()= msg->pose.position.x;
+        position.y()= msg->pose.position.y;
+
+        if(boost::geometry::within(position, current.polygon2d())) //If vehicle is inside current_lanelet, there is no crosstrack error
+        {
+            return false;
```

Comment:
now before this, we could reset our counter to 0. `cte_count_ = 0` because we could be carrying counts generated from noise in the lanelets long ago.